query/src/interpreters/interpreter_insert.rs

```diff
@@ -83,10 +99,25 @@ impl InsertInterpreter {
                 }
                 pipeline.add_pipe(builder.finalize());
             }
-            InsertInputSource::StreamingWithFormat(_) => {}
-            InsertInputSource::SelectPlan(_) => {
-                // // todo!();
-                // Some(BlocksSource::create(vec![]))
+            InsertInputSource::StreamingWithFormat(_) => {
+                pipeline.add_pipe(
+                    self.clone()
+                        .get_box()
+                        .as_mut()
+                        .source_pipe_builder
+                        .take()
+                        .ok_or_else(|| ErrorCode::EmptyData("empty source pipe builder"))?
+                        .finalize(),
+                );
+            }
+            InsertInputSource::SelectPlan(plan) => {
+                let builder = QueryPipelineBuilder::create(self.ctx.clone());
```

Comment:
duplicate codes with `SelectInterpreter`.

We can create `SelectInterpreter` and pub the method `create_new_pipeline`.